### ISSUE TITLE
How do I reject a file with a error message in a Custom Plugin from DID_LOAD_ITEM?

### ISSUE THREAD

**Issue by metadevj** [pqina/filepond#750]
Hi, 

I'm running ML (Tensorflow) from within addFilter('DID_LOAD_ITEM', function (item, _ref3) {}); in my custom plugin and I'm trying to return an error to the user when the ML doesn't find an object.

```
  addFilter('DID_LOAD_ITEM', function (item, _ref3) {

    var query = _ref3.query;
    return new Promise(function (resolve, reject) {

     .....
 if (noItemFound === true) {
                reject({
                  status: {
                    main: query('GET_NO_OBJECT_ERROR_MSG'),
                    sub: query('GET_NO_OBJECT_ERROR_MSG')
                  }
                });
                return;
              }  else {

     .....
                  item.setMetadata('crop', {
                    center: {
                      x: centerX,
                      y: centerY
                    },

                    flip: {
                      horizontal: false,
                      vertical: false
                    },
                    rotation: 0,
                    zoom: calcZoom,
                    aspectRatio: objectAspectRatio
                  });
     .....
               return resolve(item);
}
     .....
    });
  });


```
But it's not working and the user doesn't see the error message.  But it does work in LOAD_FILE.

As far as I can tell I can only run item.setMetadata('crop', {}) from DID_LOAD_ITEM and not LOAD_FILE.

So what is my options if I need to .setMetadata('crop', {}) and return a error message if something is not found?

Any ideas?

Thanks,

Metadevj

**Comment by rikschennink** [comment]
Maybe you can add an entry in metadata and then check for that entry in LOAD_FILE?

**Comment by rikschennink** [comment]
Just published `4.29.0` which adds rejecting `DID_LOAD_ITEM` with error message like shown below

```js
const customPlugin = ({ addFilter, utils }) => {
    const { Type } = utils;

    // intercept load item
    addFilter(
        'DID_LOAD_ITEM',
        (item, { query }) =>
            new Promise((resolve, reject) => {
                if (!query('GET_ALLOW_CUSTOM_PLUGIN')) resolve();

                reject({
                    error: 'Error code',
                    status: {
                        main: 'UI error main',
                        sub: 'UI error sub',
                    },
                });
            })
    );

    return {
        options: {
            // Enable or disable plugin
            allowCustomPlugin: [true, Type.BOOLEAN],

            // other options here
            //...
        },
    };
};
```

**Comment by metadevj** [comment]
Thank you very much, I've tested the functionality and it works as excepted.
